### PR TITLE
Rashmi/node health perf

### PR DESCRIPTION
**Pull request by rashmichandrashekar**
First stab at health data collection

HealthConfig in daemonset looks like this:
{
  "NodeCpuMonitor": {
    "PassPercentage": 2.0,
    "FailPercentage": 2.0,
    "MonitorTimeOut":  5
  },
  "NodeMemoryRssMonitor": {
    "PassPercentage": 2.0,
    "FailPercentage": 2.0,
    "MonitorTimeOut": 5
  },
  "DockerMonitor": {
    "MonitorTimeOut" :  3
  }
}

### CODE REVIEW

File: source/code/plugin/KubernetesApiClient.rb [hunk 1]
```diff
@@ -2,474 +2,480 @@
...
-                    if File.exist?(@@TokenFileName) && File.readable?(@@TokenFileName)
-                        @@TokenStr = File.read(@@TokenFileName).strip
-                        return @@TokenStr
-                    else
-                       @Log.warn("Unable to read token string from #{@@TokenFileName}: #{error}")
-                       return nil
-                    end
-                end
-            end
+    def getClusterRegion
```
Comment (rashmichandrashekar) [inline source/code/plugin/KubernetesApiClient.rb]: Changes for this file includes addition of this method and formatting

File: source/code/plugin/in_cadvisor_perf.rb [hunk 2]
```diff
@@ -2,88 +2,89 @@
...
-            @@istestvar = ENV['ISTEST']
-            if (!@@istestvar.nil? && !@@istestvar.empty? && @@istestvar.casecmp('true') == 0 && eventStream.count > 0)
-              $log.info("cAdvisorPerfEmitStreamSuccess @ #{Time.now.utc.iso8601}")
-            end
-            rescue  => errorStr
-            $log.warn "Failed to retrieve cadvisor metric data: #{errorStr}"
-            $log.debug_backtrace(errorStr.backtrace)
+
+        router.emit_stream(@tag, eventStream) if eventStream
+        router.emit_stream(@healthtag, eventStream) if eventStream
```
Comment (rashmichandrashekar) [inline source/code/plugin/in_cadvisor_perf.rb]: changes to this file only includes addition of stream with health tag and formatting

File: installer/conf/container.conf [hunk 3]
```diff
@@ -31,6 +31,30 @@
   log_level info
 </filter>
 
+<source>
+  type dockerhealth
```
Comment (vishiy) [inline installer/conf/container.conf]: are we sure that we are releasing this as part of next agent release ? if not, we should not merge this until we know ux is ready and all works/make sense.

Comment (vishiy) [reply installer/conf/container.conf]: huh i see that its a standup PR and took changes from ci_feature into your branch. makes sense now :)

File: source/code/plugin/filter_health_cpu_memory.rb [hunk 4]
```diff
@@ -66,7 +68,8 @@ def start
       @@cpu_limit = 0.0
       @@memory_limit = 0.0
       begin
-        nodeInventory = JSON.parse(KubernetesApiClient.getKubeResourceInfo("nodes").body)
+        nodeInventory = JSON.parse(KubernetesApiClient.getKubeResourceInfo("nodes?fieldSelector=metadata.name%3D#{@@hostName}").body)
```
Comment (r-dilip) [inline source/code/plugin/filter_health_cpu_memory.rb]: %3D ? Have u checked the fieldSelector=XXX works?

**Pull request closed without merging**